### ISSUE TITLE
[Question] Feature importance for Multi-output Regression

### ISSUE THREAD

**Issue by GalKepler** [automl/auto-sklearn#1630]
# TLDR:
It's more of a theoretical question - How can I interpret the features' importance of a multi-output regressor?

# Explanation
I'm fitting a multi-output regressor to predict several (274) targets.
Presumably, features' importance would vary across targets. However, following the permutation importance paradigm (as suggested [here](https://automl.github.io/auto-sklearn/master/examples/40_advanced/example_inspect_predictions.html#compute-permutation-importance-part-1), I get a single distribution of importances (one for each repeat) for each feature.

The results of this query fit very well with my prior knowledge of the data. Still, I was wondering what's the "correct" interpretation of it for a multi-output regression problem.

Thanks in advance!

**Comment by eddiebergman** [comment]
Hi @GalKepler,

The short answer is that I don't know how to do this properly and is more outside the scope of auto-sklearn as a whole. My guess at the interpretation is just how important is each feature with respect to some `metric(y_pred, y_test)`, regardless of whether it's multioutput or single output regression. 

One possible solution is to dig into the line:
```python
r = permutation_importance(automl, X_test, y_test, n_repeats=10, random_state=0)
```

Here I'm assuming `X_test` is somehow being used to generate predictions `y_pred` from `automl` to get permutation importance for `y_test`. If you could go in and modify the permutation importance to act on just one column, i.e. predict on `X_test` as they do, select `y_pred[0]` and compare to `y_test[0]`, then you might be able to iterate through each column, likely at a huge expense of time.

I'm going to close this though, it's sort of outside the scope of auto-sklearn and more a research topic or better for library that handles permutation importance!

Best
